webhook.go

```diff
@@ -6,6 +6,5 @@ import "github.com/gin-gonic/gin"
 type Webhooker interface {
 	Plugin
 	// RegisterWebhook is called for plugins to create their own handler.
-	// Plugins can call mux.BasePath() to acquire their custom handler base path.
-	RegisterWebhook(mux gin.RouterGroup)
+	RegisterWebhook(basePath string, mux *gin.RouterGroup)
```

Comment:
For what was this good again? If the plugin needs the url inside a route, then github.com/gotify/location.Get(*gin.Context) could be used.

Reply:
The plugin might need this information to assemble the full webhook URL in Displayer

Reply:
So in GetDisplay something like this would be done?

```go
type MyPlugin struct {
    baseURL string
}

func (p MyPlugin ) GetDisplay(location *url.URL) string {
     return "your url is " + location.Scheme + "://" + location.Host + "/" + p.baseURL + "/somewebhook"
}
```

Reply:
Yeah

Reply:
Oke, this probably could be done more plugin friendly, but it also would reduce the power of a plugin author, so I'm k with it.

Reply:
I don't think the plugin should be able to manipulate the `gin.Context` in a `Displayer` call. This might break the integrity of the core API.